code/model/Subsite.php

```diff
@@ -697,20 +753,20 @@ public function allowedThemes()
     {
         if ($themes = $this->stat('allowed_themes')) {
             return ArrayLib::valuekey($themes);
-        } else {
-            $themes = array();
-            if (is_dir(THEMES_PATH)) {
-                foreach (scandir(THEMES_PATH) as $theme) {
-                    if ($theme[0] == '.') {
-                        continue;
-                    }
-                    $theme = strtok($theme, '_');
-                    $themes[$theme] = $theme;
+        }
+
+        $themes = [];
+        if (is_dir(THEMES_PATH)) {
+            foreach (scandir(THEMES_PATH) as $theme) {
+                if ($theme[0] == '.') {
+                    continue;
                 }
-                ksort($themes);
+                $theme = strtok($theme, '_');
+                $themes[$theme] = $theme;
             }
-            return $themes;
+            ksort($themes);
         }
+        return $themes;
```

Comment:
I wonder if we could use `ThemeManifest` to replace this logic?

Reply:
Might not work, since ThemeManfest is really the manifest for the $default theme (not so much the non-default themes) but see what milage you get out of it.